src/sentry_plugins/trello/plugin.py

```diff
@@ -0,0 +1,278 @@
+        trello_client = TrelloApiClient(config["key"], config["token"])
+        try:
+            trello_client.get_organization_options()
+        except Exception as e:
+            raise PluginError(self.message_from_error(e))
+        return config
+
+    def get_group_urls(self):
+        """
+        Return the URLs and the matching views
+        """
+        return super(TrelloPlugin, self).get_group_urls() + [
+            url(
+                r"^options",
+                IssueGroupActionEndpoint.as_view(view_method_name="view_options", plugin=self),
+            ),
+            url(
+                r"^autocomplete",
+                IssueGroupActionEndpoint.as_view(view_method_name="view_autocomplete", plugin=self),
+            ),
+        ]
+
+    def is_configured(self, request, project, **kwargs):
+        return all((self.get_option(key, project) for key in ("token", "key")))
+
+    # used for boards and lists but not cards (shortLink used as ID for cards)
+    def map_to_options(self, items):
+        return [(item["id"], item["name"]) for item in items]
+
+    def get_new_issue_fields(self, request, group, event, **kwargs):
+        """
+        Return the fields needed for creating a new issue
+        """
+        fields = super(TrelloPlugin, self).get_new_issue_fields(request, group, event, **kwargs)
+        client = self.get_client(group.project)
+        organization = self.get_option("organization", group.project)
+
+        boards = client.get_boards(organization)
+        board_choices = self.map_to_options(boards)
+
+        return fields + [
+            {
+                "name": "board",
+                "label": "Board",
+                "type": "select",
+                "choices": board_choices,
+                "readonly": False,
+                "required": True,
+            },
+            {
+                "name": "list",
+                "depends": ["board"],
+                "label": "List",
+                "type": "select",
+                "has_autocomplete": False,
+                "required": True,
+            },
+        ]
+
+    def get_link_existing_issue_fields(self, request, group, event, **kwargs):
+        """
+        Return the fields needed for linking to an existing issue
+        """
+        return [
+            {
+                "name": "issue_id",
+                "label": "Card",
+                "type": "select",
+                "has_autocomplete": True,
+                "required": True,
+            },
+            {
+                "name": "comment",
+                "label": "Comment",
+                "default": absolute_uri(
+                    group.get_absolute_url(params={"referrer": "trello_plugin"})
+                ),
+                "type": "textarea",
+                "help": ("Leave blank if you don't want to " "add a comment to the Trello card."),
+                "required": False,
+            },
+        ]
+
+    def get_client(self, project):
```

Comment:
This should probably be memoized.

Reply:
Don't think this is a good idea, discussed in other comments